docker/entrypoint.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# For systems without journald.
+mkdir -p /var/log/journal
+
+if [ -z "${METADATA_AGENT_URL:-}" -a -n "${METADATA_AGENT_HOSTNAME:-}" ]; then
+  METADATA_AGENT_URL="http://${METADATA_AGENT_HOSTNAME}:8000"
+fi
+if [ -n "$METADATA_AGENT_URL" ]; then
+  sed -i "s,http://local-metadata-agent.stackdriver.com:8000,$METADATA_AGENT_URL," \
+    /etc/google-fluentd/google-fluentd.conf
+fi
+
+# This docker image supports sending either a flag, or a command, as the 
+# docker command. When a flag is sent, it will be passed on to the fluentd
+# process. Anything else will be interpreted as a command to be run.
+#
+# Passing a flag.
+# $ docker run -it {image:tag} -o /var/log/google-fluentd.log
+#
+# Passing a command.
+# $ docker run -it {image:tag} /bin/bash
+#
+# Default behavior uses CMD defined in Dockerfile.
```

Comment:
What's counterintuitive is that the CMD is passed into the `else` branch below... But I don't think it matters much to the user.

Reply:
Are you suggesting to remove this final example?

Reply:
No, just musing...